src/compas_ui/project.py

```diff
@@ -0,0 +1,105 @@
+class Project(object):
+
+    def __init__(self, ui):
+        self.stream_id = None
+        self.name = "Untitled"
+        self._ui = ui
+
+    def __getstate__(self):
+        return self.state
+
+    def __setstate__(self, state):
+        self.state = state
+
+    @property
+    def ui(self):
+        if not hasattr(self, '_ui') or self._ui is None:
+            from compas_ui.ui import UI
+            self._ui = UI()
+        return self._ui
+
+    @property
+    def state(self):
+
+        objects = []
+        data_dict = {}
+        for obj in self.ui.scene.objects:
+            objects.append(obj.stream_id)
+            data = obj.item
+            guid = str(data.guid)
+            if guid not in data_dict:
+                if hasattr(data, 'stream_id'):
+                    data_dict[guid] = data.stream_id
+                else:
+                    data_dict[guid] = None
+
+        return {
+            "name": self.name,
+            "stream_id": self.stream_id,
+            "session": self.ui.session.data,
+            "data": list(data_dict.values()),
+            "objects": objects,
+            "scenes": [self.ui.scene.stream_id]
+        }
+
+    @state.setter
+    def state(self, state):
+        self.name = state['name']
+        self.stream_id = state['stream_id']
```

Comment:
this seems to not recreate the state of the object

Reply:
regardless of the speckle part, getting and setting state should be compatible and reciprocal